### ISSUE TITLE
[Feat]: Fetch ForeignKey Data

### ISSUE THREAD

**Issue by GeekMasher** [42ByteLabs/geekorm#14]
### What new feature are you looking for / expect?

```rust
pub struct Posts {
    pub id: PrimaryKey,
    title: String,

    #[geekorm(foreign_key = "Users.id")]
    user: ForeignKey<i32, Users>,

    #[geekorm(foreign_key = "Images.id")]
    image: ForeignKey<i32, Images>
}

impl Post {
    pub fn fetch_user(&self) -> Result<Users> {
         // where query...
    }
 
    pub fn fetch_image(&self) -> Result<Images> {
         // where query...
    }
}

```

### Expectations

Required / Manditory Feature

**Comment by GeekMasher** [comment]
Feature added in #17

**Comment by GeekMasher** [comment]
Released in v0.2.0
